### DISCUSSION TITLE
shortcodes in HYAS

### DISCUSSION THREAD

**Discussion by technosprout** [gethyas/hyas#431] [Q&A]
Hello there

What is the proper methods to use shortcodes in HYAS?
- Is there a mount code line in module.toml?
- where would one place the shortcode .html files?

**Comment by salim-b** [comment]
Depends on what version of Doks you're using / how you set up Doks.

Generally, shortcodes are templates and live under Hugo's `layouts` component folder, under `layouts/shortcodes/` to be precise.

If you created your project with the [create-hyas CLI](https://github.com/gethyas/create-hyas) (`npm create hyas@latest my-doks-project -- --template doks`), then [this `module.toml` config](https://github.com/gethyas/create-hyas/blob/main/template-doks/config/_default/module.toml) should apply. It includes an entry

```toml
[[mounts]]
  source = "layouts"
  target = "layouts"
```

that means to mount the `layouts/` folder on the top level of your project to `layouts/` in Hugo's [virtual union filesystem](https://gohugo.io/getting-started/directory-structure/#union-file-system). Simple as that. If you have that entry (*below* other entries with `target = "layouts"`, so it has precedence), you can put your custom shortcode files into `layouts/shortcodes/`.
